lib/DAVClient.ts

```diff
@@ -40,4 +40,7 @@ export class DAVClient {
   public requestText(options: RequestOptions): Promise<string> {
     return this.#httpClient.requestText(this.buildRequestOptions(options));
   }
+  public request(options: RequestOptions): Promise<Response> {
+    return this.#httpClient.request(this.buildRequestOptions(options));
+  }
```

Comment:
- `requestText` is for when the response body is just text.
- `requestJson` is for when the response body is json.

In the case when the response body empty, I guess you can use `requestJson`, so no need to add this method.

Reply:
it was the review of @rezk2ll  to use request , in all cases we need to add this method , if not why it is added in HttpClient interface

Reply:
You don't need to use a new method when there's already a method that can do the same job (which is `requestJson`).